File: internal/content/testdata/bench/javascript/nc-koko-bananas.js
function min_eating_speed(piles, h) {
    let lo = 1, hi = Math.max(...piles);
    let ans = hi;
    while (lo <= hi) {
        const mid = Math.floor((lo + hi) / 2);
        let hours = 0;
        for (const p of piles) hours += Math.ceil(p / mid);
        if (hours <= h) {
            ans = mid;
            hi = mid - 1;
        } else {
            lo = mid + 1;
        }
    }
    return ans;
}

File: internal/content/testdata/bench/go/nc-koko-bananas.go
func min_eating_speed(piles []int, h int) int {
	lo, hi := 1, 0
	for _, p := range piles {
		if p > hi {
			hi = p
		}
	}
	ans := hi
	for lo <= hi {
		mid := (lo + hi) / 2
		hours := 0
		for _, p := range piles {
			hours += (p + mid - 1) / mid
		}
		if hours <= h {
			ans = mid
			hi = mid - 1
		} else {
			lo = mid + 1
		}
	}
	return ans
}
